src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -84,40 +48,57 @@ private static bool TryGetStringValue(string switchName, string envVariable, [No
             return true;
         }
 
-        private static string CreateLibraryName(ReadOnlySpan<char> baseName, ReadOnlySpan<char> suffix, ReadOnlySpan<char> extension, ReadOnlySpan<char> version, bool versionAtEnd = false)
+        private static void LoadAppLocalIcu(string icuSuffixAndVersion, bool suffixWithSeparator = false)
         {
-            int length = baseName.Length + suffix.Length + version.Length + extension.Length;
-
-            // We validate that suffix and version are not larger than 53 characters.
-            Span<char> result = stackalloc char[length];
-            baseName.CopyTo(result);
-
-            Span<char> secondPart = result.Slice(baseName.Length);
-            suffix.CopyTo(secondPart);
+            ReadOnlySpan<char> icuSuffix = default;
+            ReadOnlySpan<char> version = default;
 
-            Span<char> middle = secondPart.Slice(suffix.Length);
-
-            if (!versionAtEnd)
+            // Custom built ICU can have a suffix on the name, i.e: libicuucmyapp.so.67.1
+            // So users would set the runtime switch as: myapp:67.1
+            int indexOfSeparator = icuSuffixAndVersion.IndexOf(':');
+            if (indexOfSeparator >= 0)
             {
-                version.CopyTo(middle);
+                icuSuffix = icuSuffixAndVersion.AsSpan().Slice(0, indexOfSeparator);
```

Comment:
I think this is fine but just want to confirm when ':' exist at the beginning of the string, this slicing will produce empty span (with length zero). right?

Reply:
Right, I tested it and it will be that way, so it is the equivalent of not passing a suffix.